stl/inc/tuple

```diff
@@ -343,6 +369,32 @@ public:
         tuple(const pair<_First, _Second>&& _Right) noexcept(
             _Tuple_nothrow_constructible_v<tuple, const _First, const _Second>) // strengthened
         : tuple(_Unpack_tuple_t{}, _STD move(_Right)) {}
+
+#ifdef __cpp_lib_concepts
+    template <class _Other, class _Indices = index_sequence_for<_Rest...>>
+    static constexpr bool _Can_construct_from_tuple_like_v = false;
+
+    template <_Tuple_like _Other, size_t... _Indices>
+    static constexpr bool _Can_construct_from_tuple_like_v<_Other, index_sequence<_Indices...>> =
+        _Different_from<_Other, tuple> && !_Is_subrange_v<remove_cvref_t<_Other>>
+        && 1 + sizeof...(_Rest) == tuple_size_v<remove_cvref_t<_Other>>
+        && (is_constructible_v<_This, decltype(_STD get<0>(_STD declval<_Other>()))> && ...
+            && is_constructible_v<_Rest, decltype(_STD get<_Indices + 1>(_STD declval<_Other>()))>) //
+        &&(sizeof...(_Rest) != 0 || (!is_convertible_v<_Other, _This> && !is_constructible_v<_This, _Other>) );
+
+    template <class _Other, class _Indices = index_sequence_for<_Rest...>>
+    static constexpr bool _Is_tuple_like_constructor_explicit_v = false;
+
+    template <_Tuple_like _Other, size_t... _Indices>
+    static constexpr bool _Is_tuple_like_constructor_explicit_v<_Other, index_sequence<_Indices...>> =
+        !(is_convertible_v<decltype(_STD get<0>(_STD declval<_Other>())), _This> && ...
+            && is_convertible_v<decltype(_STD get<_Indices + 1>(_STD declval<_Other>())), _Rest>);
+
+    template <_Tuple_like _Other>
+        requires _Can_construct_from_tuple_like_v<_Other>

```

Comment:
Is this redundantly constrainted? `_Can_construct_from_tuple_like_v` is always `false` when `_Other` doesn't satisify `_Tuple_like`. Same for the `operator=`.

Reply:
*Resolving since `_Can_construct_from_tuple_like_v` and others were removed.*